internal/server/spanner/datasource.go

```
@@ -91,6 +94,14 @@ func (sds *SpannerDataSource) Observation(ctx context.Context, req *pbv2.Observa
 
 	observations = filterObservationsByDateAndFacet(observations, date, req.Filter)
 
+	if len(observations) > shared.MaxSeries {
+		return nil, status.Errorf(
+			codes.Internal,
+			"Stop processing large number of concurrent observation series: %d",
```

Should we return an error message that helps the user take an alternative action? Something along the lines of:

"Your request exceeds the allowed limit on the number of time series processed: x > y. Please try reducing the number of variables or entities in your request."

---

Good idea, thanks! Updated